DeploymentAgentReconfigurationScripts/TFS2018 Update2/HardwareMoveRemoveOldAgent.ps1

```diff
@@ -0,0 +1,178 @@
+
+Inputs:
+1. targetTFSUrl : New TFS url after hardware move.
+2. patToken : PAT token from new TFS with Deployment group manage scope (at least)
+3. agentDownloadUrl: this is optional, user can specify if she wants any specific agent version to be installed.
+4. existingAgentFolder: The script will Auto-detect the agent folder if it was running as windows service. User need to pass the folder path otherwise.
+5. action: By default, the script will not do any update operation, it will just print what changes will happen after the script is applied. User need to set action parameter to 'apply' to execute actual steps.
+
+Output:
+If action is set to apply, this script will
+1. Delete the existing unusable offline agent and re-configure a new agent in with same name and properties including tags.
+2. All existing deployment history will be deleted. 
+
+Example usage:
+.\HardwareMoveRemoveOldAgent.ps1 -targetTFSUrl <newTfsUrl> -patToken <PAT-for-new-TFS>
+.\HardwareMoveRemoveOldAgent.ps1 -targetTFSUrl <newTfsUrl> -patToken <PAT-for-new-TFS> -existingAgentFolder <AgentFoilder (C:\vstsagents\A1)>
+.\HardwareMoveRemoveOldAgent.ps1 -targetTFSUrl <newTfsUrl> -patToken <PAT-for-new-TFS> -existingAgentFolder <AgentFoilder (C:\vstsagents\A1)> -action 'apply'
+#>
+
+param([string]$targetTFSUrl,
+      [string]$patToken,
+      [string]$existingAgentFolder = "",
+      [string]$agentDownloadUrl = 'https://vstsagentpackage.azureedge.net/agent/2.131.0/vsts-agent-win-x64-2.131.0.zip',
+      [string]$action = "PrintEffect")
+
+$ErrorActionPreference="Stop"
+
+# Basic input validations
+If (-NOT ([Security.Principal.WindowsPrincipal][Security.Principal.WindowsIdentity]::GetCurrent() ).IsInRole( [Security.Principal.WindowsBuiltInRole] �Administrator�)){ 
+    throw "Run command in an administrator PowerShell prompt"
+};
+
+If ($PSVersionTable.PSVersion -lt (New-Object System.Version("3.0"))){
+    throw "The minimum version of Windows PowerShell that is required by the script (3.0) does not match the currently running version of Windows PowerShell."
+}
+
+
+# Auto detect existing agent folder
+if ($existingAgentFolder -eq ""){
+    $ServiceNamePrefix = 'vstsagent*'
+    $agentServices = Get-Service -Name $ServiceNamePrefix
+    if ($agentServices.Count -eq 1){
+        $serviceName = $agentServices[0].Name;
+        $service = get-wmiobject -query "select * from win32_service where name = `'$serviceName`'";
+        $serviceExePath = $service.pathname;
+        # $serviceExePath will have value like "C:\vstsAgent2\A8\bin\AgentService.exe"
+        $existingAgentFolder = $serviceExePath.Substring(1, $serviceExePath.Length - 22);
```

Comment:
Try to use Split-Path in order to get to the parent. Better than hard coding 22  IMO.